docs: Add React quick start

sumitsaurabh927 opened the pull request:
### What change does this PR introduce?

Adds a quick start guide for React and NextJS
### Why was this change needed?
There was no quick start guide for React and NextJS

### Other information (Screenshots)

<!-- Add notes or any other information here -->
<!-- Also add all the screenshots which support your changes -->

p-fernandez commented on .vscode/settings.json:
```diff
@@ -16,5 +16,8 @@
   },
   "cSpell.words": ["mantine"],
-  "vsicons.presets.nestjs": true
+  "vsicons.presets.nestjs": true,
+  "[markdown]": {
+    "editor.defaultFormatter": "DavidAnson.vscode-markdownlint"
+  }
```
Please rollback this change.

sumitsaurabh927 replied:
Resolved [here](https://github.com/novuhq/novu/pull/3590/commits/16ebbf0d257984db468396ca8f54b414ef322ea5)

p-fernandez commented on docs/docs/overview/quickstart/get-started-with-React.md:
Please lowercase the filename to `get-started-with-react.md`.

sumitsaurabh927 replied:
Done [here](https://github.com/novuhq/novu/pull/3590/commits/16ebbf0d257984db468396ca8f54b414ef322ea5)

unicodeveloper replied:
@sumitsaurabh927 did you change the file name to lower case because I can't see it? The commit here refers to vscode settings you reverted.

p-fernandez commented on docs/docs/overview/quickstart/get-started-with-React.md:
```diff
@@ -0,0 +1,230 @@
+- A Novu account. [Sign up for free](http://web.novu.co) if you don’t have one yet.
+- A working React development environment.
+
+You can find the code for this project [here](https://github.com/sumitsaurabh927/react-fe-for-quickstart)
```
We would need to move the project code under our organization repositories.

sumitsaurabh927 replied:
Done here d7c2076

p-fernandez commented on docs/docs/overview/quickstart/get-started-with-React.md:
```diff
@@ -0,0 +1,230 @@
+
+## Install Novu React Notification Center Package
+
+The Novu Vue package provides a React component library that you can use to add a fully functioning notification center into your React application.
```
```suggestion
The Novu React package provides a React component library that you can use to add a fully functioning notification center into your React application.
```

sumitsaurabh927 replied:
Thanks for pointing this out! 🙏

I've fixed it in the latest commit - 781d0dc

p-fernandez commented on docs/docs/overview/quickstart/get-started-with-React.md:
````diff
@@ -0,0 +1,230 @@
+Create a new React app by running the following command in your terminal:
+
+```bash
+npx create-react-app notifications-app
````
`create-react-app` has been deprecated so I do not think it is a good idea to recommend its use. I think this section should be skipped for now.

sumitsaurabh927 replied:
Okay. Thank you for this!
I've removed this section in this commit:
2a8b842

For future apps, what tool shall I use in place of create-react-app?

p-fernandez replied:
I am not very aware what is the best React app boilerplate right now. Sorry as I can't be helpful on that part. If I discover which one will let you know.

p-fernandez commented on docs/docs/overview/quickstart/get-started-with-React.md:
```diff
@@ -0,0 +1,230 @@
+    e.preventDefault();
+    const response = await axios.post('http://localhost:3000/sendNotif/create', { description });
+    setDescription('');
+    // console.log(response);
```
```suggestion
```
🗑️

p-fernandez commented on docs/docs/overview/quickstart/get-started-with-React.md:
```diff
@@ -0,0 +1,230 @@
+  }
+  const onClickHandler = async (e) => {
+    e.preventDefault();
+    const response = await axios.post('http://localhost:3000/sendNotif/create', { description });
```
I do not understand this endpoint. Is it a custom made by you in an own API?

sumitsaurabh927 replied:
Yes. I made this end point in the back-end I wrote. Shall I make any changes here?

unicodeveloper replied:
@sumitsaurabh927 please remove the entire onclickHandler function.

sumitsaurabh927 replied:
done in the latest commit

p-fernandez commented on docs/docs/overview/quickstart/get-started-with-React.md:
```diff
@@ -0,0 +1,230 @@
+
+## Trigger a notification
+
+We can trigger a notification by simply using Novu’s trigger method as shown below:
```
I find a bit confusing to introduce here the trigger of a notification using the Node package. Not sure if it is the right direction to go as might bias users to do it like this when they have other options to trigger the notifications (other SDKs, API, etc).
@unicodeveloper what do you think?

sumitsaurabh927 replied:
@unicodeveloper Shall I move it to a codesandbox?

unicodeveloper replied:
@sumitsaurabh927 no need, you can use the multi-language template syntax here. So that it can show how to trigger notification from multiple languages and SDK

sumitsaurabh927 replied:
But this quick start is for React. Would it still make sense or shall I let this be (to trigger notifs in React apps) and have a callout stating that it can be triggered across multiple languages and SDKs?

unicodeveloper replied:
Whether it's react, vue, angular, the triggering of notification happens from the backend. And it can be any backend. Node, PHP, Python etc.

unicodeveloper replied:
To make it simple, just move it to a codesandbox, or use an example from the ones we already have in codesandbox

sumitsaurabh927 replied:
Done in latest commit!

p-fernandez reviewed (changes requested):
Thank you for your contribution. Some things to change and there are some open questions.

sumitsaurabh927 commented:
@unicodeveloper please also review the NextJS quickstart here 
4ba9c70

sumitsaurabh927 commented:
https://github.com/novuhq/novu/assets/62152915/4aecdf53-91ac-41f0-a4a5-e7f8be70f72f

@unicodeveloper as you had asked!

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-React.md:
```diff
@@ -0,0 +1,230 @@
+        </PopoverNotificationCenter>
+      </NovuProvider>
+
+      <form onSubmit={onClickHandler}>
+        <input
+          placeholder="Enter notification text"
+          value={description}
+          onChange={(e) => setDescription(e.target.value)}
+        />
+        <button type="submit">Send</button>
+      </form>
```
Remove this as well

sumitsaurabh927 replied:
done!

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-React.md:
````diff
@@ -0,0 +1,230 @@
+
+With Novu, you can create a subscriber using any of its SDKs (Node.js, PHP, .NET, Go, Ruby, Python and Kotlin). The code to create a subscriber in Novu is:
+
+```jsx
````
Starting from here, all the node code examples should show in both Node, and PHP. 

Check out the code sample here https://docs.novu.co/platform/preferences/#get-subscriber-preferences

sumitsaurabh927 commented:
https://github.com/novuhq/novu/assets/62152915/0806af9b-7085-4931-9c3d-149dec9a9b4f

@unicodeveloper please review and merge

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-React.md:
```diff
@@ -0,0 +1,247 @@
+Before you can use Novu in your react app, you'll need two things:
+
+1. Create a notification template to use for sending notifications, and
+2. Create a subscriber - recipients of Notifications.
```
Notifications => notifications

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-React.md:
```diff
@@ -0,0 +1,247 @@
+
+These are the steps to create a notification template:
+
+1. Click **Notifications** on the left sidebar of your Novu dashboard.
```
**Notifications** => **Workflows**

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-React.md:
```diff
@@ -0,0 +1,247 @@
+These are the steps to create a notification template:
+
+1. Click **Notifications** on the left sidebar of your Novu dashboard.
+2. Click the **"Create Workflow"** button on the top right.
```
**"Create Workflow"** => **Create Workflow**

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-React.md:
```diff
@@ -0,0 +1,247 @@
+
+1. Click **Notifications** on the left sidebar of your Novu dashboard.
+2. Click the **"Create Workflow"** button on the top right.
+3. The name of a new notification template is currently **"Untitled"**. Rename it to a more suitable title.
```
notification template => notification workflow

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-React.md:
```diff
@@ -0,0 +1,247 @@
+
+## Create a notification template
+
+Before triggering a notification, we need to create a notification template. A template is like a blueprint that all the notifications are supposed to follow.
```
notification template => notification workflow

unicodeveloper replied:
A template => A workflow

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-React.md:
```diff
@@ -0,0 +1,247 @@
+
+> The recipients of a triggered notification are called **subscribers**.
+
+The template includes the following:
```
The template => The workflow

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-React.md:
```diff
@@ -0,0 +1,247 @@
+
+The template includes the following:
+
+- Notification template name and Identifier
```
Notification template name => Notification workflow name

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-React.md:
```diff
@@ -0,0 +1,247 @@
+1. Click **Notifications** on the left sidebar of your Novu dashboard.
+2. Click the **"Create Workflow"** button on the top right.
+3. The name of a new notification template is currently **"Untitled"**. Rename it to a more suitable title.
+4. Select **"In-App"** as the channel you want to add.
```
**"In-App"** => **In-App**

sumitsaurabh927 commented:
Hey @unicodeveloper I've addressed all your feedback points and made changes accordingly. I've also gone ahead and replaced 'template' with 'workflow' as it should be. 
Please take a look!

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-React.md:
```diff
@@ -0,0 +1,247 @@
+await novu.subscribers.identify('123', {
+  firstName: 'Sumit',
+  lastName: 'Saurabh',
+  returnUser: true,
```
Is returnUser really an attribute? I have never seen this before. @jainpawan21 @sumitsaurabh927 just confirming to be sure.

sumitsaurabh927 replied:
No, good catch!

I had done this because I was trying to see if the created subscriber is returned (in postman) and should have removed it.

My bad!

Removing it from the guide.

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-React.md:
```diff
@@ -0,0 +1,247 @@
+
+$novu->updateSubscriber('111', [
+    // new email
+    'email' => 'john@domain.com',
+    // new phone
+    'phone' => '+19874567832',
+])->toArray();
```
Here should also be `firstName` and `lastName` like you have in the Node equivalent.

sumitsaurabh927 replied:
Done ✅

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-React.md:
```diff
@@ -0,0 +1,247 @@
+
+$novu = new Novu('<NOVU_API_KEY>');
+
+$novu->updateSubscriber('111', [
```
`111` should be `123` like you have in the Node equivalent. 

The goal is to be able to do the same things in different languages so it has to accomplish the same purpose with the same data / parameters as well.

sumitsaurabh927 replied:
Done 🚀

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-React.md:
```diff
@@ -0,0 +1,247 @@
+
+## Trigger a notification
+
+We can trigger a notification by simply running the code sandbox below with the correct credentials.
```
code sandbox => codesandbox

sumitsaurabh927 replied:
fixed!

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-React.md:
```diff
@@ -0,0 +1,247 @@
+    'subscriberId' => '123',
+    'firstName' => 'Sumit',
+    'lastName' => 'Saurabh',
+    'phone' => '+13603963366',
+    'avatar' => 'https://example.com/images/avatar.jpg',
+    'locale' => 'en',
+    'data' => [
+      'customKey1' => 'customVal1',
+      'customKey2' => 'customVal2'
+    ]
```
Please remove the phone, avatar, locale and data properties.

sumitsaurabh927 replied:
Removed 🗑️

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-nextjs.md:
```diff
@@ -0,0 +1,319 @@
+
+# NextJS Quickstart
+
+This quick start guide is written with the intention of helping you integrate Novu into your NextJS app in no time. Novu is an open-source notifications infrastructure that you can use in your app to deliver rich notifications across channels like sms, email, in-app, push, chat, etc. In this quick start, you’ll learn how to integrate email notifications into your NextJS app.
```
quick start => quickstart

unicodeveloper replied:
In this quick start => In this guide

sumitsaurabh927 replied:
addressed both

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-nextjs.md:
```diff
@@ -0,0 +1,319 @@
+
+## Create a NextJS app to get started
+
+The first step here would be to create a NextJS app. We’ll later play with Novu in this app itself. To get started, open your terminal and create a NextJS app using the following command:
```
We’ll later play with Novu in this app itself => We will play with Novu in this app later.

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-nextjs.md:
```diff
@@ -0,0 +1,319 @@
+
+For each channel, there can be only one provider active at a time. Although the exact setup process varies from provider to provider, the general flow is signing up for a provider, getting an API key from its portal, and plugging it into the Novu web portal.
+
+Once having integrated a provider, we need a notification template to send notifications. One can have dynamic data in this template if they so choose.
```
notification template => notification workflow

unicodeveloper replied:
template => workflow

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-nextjs.md:
```diff
@@ -0,0 +1,319 @@
+
+Once having integrated a provider, we need a notification template to send notifications. One can have dynamic data in this template if they so choose.
+
+In our case, we’ll have dynamic data and whatever we send as a description will be sent as an email notification. Following are the steps to create a notification template.
```
notification template => notification workflow

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-nextjs.md:
```diff
@@ -0,0 +1,319 @@
+
+In our case, we’ll have dynamic data and whatever we send as a description will be sent as an email notification. Following are the steps to create a notification template.
+
+## Creating a notification template
```
Creating a notification template => Create a Notification Workflow

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-nextjs.md:
```diff
@@ -0,0 +1,319 @@
+
+## Creating a notification template
+
+1. Click “Notifications” on the left sidebar of your Novu dashboard.
```
"Notifications" => "Workflows"

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-nextjs.md:
```diff
@@ -0,0 +1,319 @@
+1. Click “Notifications” on the left sidebar of your Novu dashboard.
+2. Click the “Create Workflow” button on the top right.
+   ![Creating a workflow in Novu dashboard](https://res.cloudinary.com/dxc6bnman/image/upload/v1686777195/guides/Untitled_c2x7mn.png)
+3. The name of a new notification template is currently "Untitled." Rename it to a more suitable title.
```
notification template => notification workflow

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-nextjs.md:
```diff
@@ -0,0 +1,319 @@
+
+A topic can have multiple subscribers associated with it. These subscribers will receive notifications whenever a notification is dispatched to the respective topic.
+
+### Creating a topic
```
Creating => Create

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-nextjs.md:
```diff
@@ -0,0 +1,319 @@
+
+> Note how the return object contains the key I’d sent from my request body. That signals successful creation!
+
+## Adding subscriber to a Topic
```
Adding => Add

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-nextjs.md:
```diff
@@ -0,0 +1,319 @@
+
+This adds the subscriber with the `subscriberID` we’d passed to the array of subscribers, which in the above case contains just one subscriber.
+
+If you check this on Postman, the array returned will contain the `subsriberID` (highlighted in the result) we’d passed in the request body signaling that it was added, as shown below:
```
subsriberID => subscriberID

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-nextjs.md:
```diff
@@ -0,0 +1,319 @@
+
+## Sending notifications to a topic
+
+Sending notifications to a topic is pretty straightforward. You just need to extract the topic key to which you want to send notifications and trigger Novu’s method on that topic key with the message in the payload:
```
We need to be careful to avoid using sentences like "pretty straightforward" because it might not be for many people.  We can reword your sentence from "Sending notifications to a topic is pretty straightforward" to "Sending notifications to a topic is not a complex task".

unicodeveloper replied:
You just need to extract => You need to extract..

sumitsaurabh927 replied:
Okay, will take this into account from now on. Thanks!

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-vanilla-js.md:
```diff
@@ -0,0 +1,315 @@
+
+# Vanilla JS Quickstart
+
+Learn how to integrate Novu into your vanilla JS app on the fly. Send notifications across channels (SMS, Email, Chat, Push) with the help of a rich, customizable, real-time In-App notification center
```
"Send notifications across channels (SMS, Email, Chat, Push) with the help of a rich, customizable, real-time In-App notification center." - This feels like  the notification center can send notifications across all these channels.

It can be reworded to something like:

Send notifications across different channels (SMS, Email, Chat, Push) and  enable real-time In-App notifications with the help of a rich and customizable Notification Center.

sumitsaurabh927 replied:
I see your point. Making changes accordingly!

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-vanilla-js.md:
```diff
@@ -0,0 +1,315 @@
+
+## Loading Notification center component from the CDN
+
+The Notification Center Web Component is a custom element that can be used in any web application. It is built by wrapping the React components and consists from its core the three pieces: NovuProvider, PopoverNotificationCenter, and NotificationBell combined together into one.
```
Please remove `It is built by wrapping the React components and consists from its core the three pieces: NovuProvider, PopoverNotificationCenter, and NotificationBell combined together into one.` .  That piece of info is not really needed here.

It consists

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-vanilla-js.md:
````diff
@@ -0,0 +1,315 @@
+
+const novu = new Novu('<YOUR_NOVU_API_KEY>');
+
+await novu.subscribers.identify('123', {
+  firstName: 'Sumit',
+  lastName: 'Saurabh',
+  returnUser: true,
+});
+```
+
+  </TabItem>
+  <TabItem value="php" label="PHP">
+
+```php
+use Novu\SDK\Novu;
+
+$novu = new Novu('<NOVU_API_KEY>');
+
+$novu->createSubscriber([
+    'subscriberId' => '123',
+    'firstName' => 'Sumit',
+    'lastName' => 'Saurabh',
+    'phone' => '+13603963366',
+    'avatar' => 'https://example.com/images/avatar.jpg',
+    'locale' => 'en',
+    'data' => [
+      'customKey1' => 'customVal1',
+      'customKey2' => 'customVal2'
+    ]
+]);
````
Like I mentioned in the other guide, the data needs to be similar/thesame in both code samples

sumitsaurabh927 replied:
done ✅

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-vanilla-js.md:
````diff
@@ -0,0 +1,315 @@
+
+const novu = new Novu('<YOUR_NOVU_API_KEY>');
+
+await novu.subscribers.update('123', {
+  firstName: 'Saurabh', // new first name
+  lastName: 'Sumit', // new last name
+});
+```
+
+  </TabItem>
+  <TabItem value="php" label="PHP">
+
+```php
+use Novu\SDK\Novu;
+
+$novu = new Novu('<NOVU_API_KEY>');
+
+$novu->updateSubscriber('111', [
+    // new email
+    'email' => 'john@domain.com',
+    // new phone
+    'phone' => '+19874567832',
+])->toArray();
````
Data needs to be thesame in both code samples

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-vanilla-js.md:
```diff
@@ -0,0 +1,315 @@
+
+These are the steps to create a notification workflow:
+
+1. Click **Notifications** on the left sidebar of your Novu dashboard.
```
Notifications => Workflows

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-vanilla-js.md:
```diff
@@ -0,0 +1,315 @@
+We can trigger a notification by simply executing the trigger code snippet we get from the Workflow tab in the Novu web dashboard as shown in the image below:
+
+![Code snippet to trigger a notification](https://res.cloudinary.com/dxc6bnman/image/upload/v1686865543/guides/Screenshot_2023-06-16_at_3.15.35_AM_wqjvzh.png)
+
```
We should also add the code snippet here for people that just want to copy straight from this guide.

sumitsaurabh927 replied:
Done ✅

unicodeveloper commented on docs/docs/overview/quickstart/get-started-with-vanilla-js.md:
```diff
@@ -292,12 +292,35 @@ The front-end as well as the back-end code for the sample application can be fou
 
 > Make sure you're executing this code with the correct credentials.
 
+This is the trigger snippet you can use to trigger notifications!
```
Any particular reason why this sentence is here? I am not sure it's necessary @sumitsaurabh927

sumitsaurabh927 replied:
Didn't want to end it with a callout so I added that. Shall I remove it?

unicodeveloper replied:
If you wrote it just before you added the trigger snipper, then that would have been great. But you wrote it after, so it's no longer necessary

unicodeveloper replied:
so please just remove it
